pkg/params/params.go

```diff
@@ -238,16 +242,60 @@ func (p *Params) ValidateStringMap(cfg map[string]string) error {
 	return nil
 }
 
+func compressAndB64Encode(s string) string {
+	// Create a new zlib.Writer, which will write to a bytes.Buffer
+	var b bytes.Buffer
+	w := zlib.NewWriter(&b)
+
+	// Write the contents of the file to the zlib.Writer
+	if _, err := io.Copy(w, strings.NewReader(s)); err != nil {
+		return "" // fmt.Errorf("failed to copy file %s to zlib.Writer: %w", path, err)
+	}
+	// Close the zlib.Writer to ensure that all data has been written
+	if err := w.Close(); err != nil {
+		return "" // fmt.Errorf("failed to close zlib.Writer: %w", err)
+	}
+	return base64.StdEncoding.EncodeToString(b.Bytes())
+}
+
+func b64DecodeAndDecompress(s string) []byte {
```

Comment:
IMHO, this should return an error, if decompression fails (and so should `CopyFromMap` below.